public/app/features/teams/state/actions.ts

```diff
@@ -111,24 +113,73 @@ export function updateTeam(name: string, email: string): ThunkResult<void> {
 export function loadTeamGroups(): ThunkResult<void> {
   return async (dispatch, getStore) => {
     const team = getStore().team.team;
-    const response = await getBackendSrv().get(`/api/teams/${team.uid}/groups`);
-    dispatch(teamGroupsLoaded(response));
+
+    if (config.featureToggles.kubernetesExternalGroupMapping) {
```

Comment:
The approach here using the `dispatch` and `initiate` is correct, but the cleaner way overall is to refactor use hooks and then we can remove the actions + reducers logic altogether. You don't have to do that here for the groups though.

I've made similar changes for the rest of the teams functionality here: https://github.com/grafana/grafana/pull/113713

I can make the changes to hooks afterward both of these PRs are merged (most likely easier at that point)